source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,927 @@
+
+If ``timeoutMS`` is set, drivers MUST use ``min(serverSelectionTimeoutMS,
+remaining timeoutMS)``, referred to as ``computedServerSelectionTimeout`` as
+the timeout for server selection and connection checkout. The server selection
+loop MUST fail with a timeout error once the timeout expires.
+
+After a server has been selected, drivers MUST use the remaining
+``computedServerSelectionTimeout`` value as the timeout for connection
+checkout. If a new connection is required, ``min(connectTimeoutMS, remaining
+computedServerSelectionTimeout)`` MUST be used as the timeout for TCP socket
+establishment. Any network requests required to create or authenticate a
+connection (e.g. HTTP requests to OCSP responders) MUST use
+``min(operationTimeout, remaining computedServerSelectionTimeout)`` as a
+timeout, where ``operationTimeout`` is the specified default timeout for the
+network request. If there is no specified default, these operations MUST use
+the remaining ``computedServerSelectionTimeout`` value. All commands sent
+during the connection’s handshake MUST use the remaining
+``computedServerSelectionTimeout`` as their ``timeoutMS`` value. Handshake
+commands MUST also set timeouts per the `Command Execution`_ section.
+
+If ``timeoutMS`` is not set and support for ``waitQueueTimeoutMS`` has not
+been removed, drivers MUST continue to exhibit the existing timeout behavior
+by honoring ``serverSelectionTimeoutMS`` for server selection and
+``waitQueueTimeoutMS`` for connection checkout. If a new connection is
+required, drivers MUST use ``connectTimeoutMS`` as the timeout for socket
+establishment and ``socketTimeoutMS`` as the socket timeout for all handshake
+commands.
+
+See `serverSelectionTimeoutMS is not deprecated`_ and `connectTimeoutMS is
+not deprecated`_.
+
+Command Execution
+~~~~~~~~~~~~~~~~~
+
+If ``timeoutMS`` is set, drivers MUST append a ``maxTimeMS`` field to
+commands executed against a MongoDB server using the 90th percentile RTT of
+the selected server. Note that this value MUST be retrieved during server
+selection using the ``servers`` field of the same `TopologyDescription
+<../server-discovery-and-monitoring/server-discovery-and-monitoring.rst#TopologyDescription>`__
+that was used for selection before the selected server's description can be
+modified. Otherwise, drivers may be subject to a race condition where a
+server is reset to the default description (e.g. due to an error in the
+monitoring thread) after it has been selected but before the RTT is
+retrieved.
+
+If the 90th percentile RTT of the selected server is less than the remaining
+timeoutMS, the value of this field MUST be ``remaining timeoutMS - 90th
+percentile RTT``. If not, drivers MUST return a timeout error without
+attempting to send the message to the server. This is done to ensure that an
+operation is not routed to the server if it will likely fail with a socket
+timeout as that could cause connection churn. The ``maxTimeMS`` field MUST be
```

Comment:
Why would it fail with a socket timeout? If remaining timeoutMS ≤ p90 RTT, that doesn't imply anything about a socket timeout or the likelihood of a network error, right?